src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,164 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.connection.ConnectionUtil;
+import mate.jdbc.exception.RecordDoesNotExistException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertQuery = "INSERT INTO manufacturers (manufacturer_name, manufacturer_country)"
+                + " VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufactureStatement = connection
+                        .prepareStatement(insertQuery, Statement.RETURN_GENERATED_KEYS)) {
+            createManufactureStatement.setString(1, manufacturer.getName());
+            createManufactureStatement.setString(2, manufacturer.getCountry());
+            createManufactureStatement.executeUpdate();
+            ResultSet generatedKeys = createManufactureStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't insert new manufacture to DB", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String selectQuery = "SELECT * FROM manufacturers WHERE manufacturer_id = ? "
+                + "AND is_deleted <> true;";
+        Manufacturer manufacturer = new Manufacturer();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturer = connection.prepareStatement(selectQuery)) {
+            if (isRecordExistInDB(id) > 0) {
+                getManufacturer.setLong(1, id);
+                ResultSet resultSet = getManufacturer
+                        .executeQuery();
+                if (!resultSet.isBeforeFirst()) {
+                    return Optional.empty();
```

Comment:
No need for isRecordExists method:

```suggestion
                getManufacturer.setLong(1, id);
                ResultSet resultSet = getManufacturer
                        .executeQuery();
                if (!resultSet.next()) {
                    return Optional.empty();
```